pkg/cosign/verify.go

```diff
@@ -697,33 +680,83 @@ func verifyInternal(ctx context.Context, sig oci.Signature, h v1.Hash,
 		return bundleVerified, nil
 	}
 
-	bundleVerified, err = VerifyBundle(ctx, sig, co.RekorClient)
+	// 2. Check the validity time of the signature.
+	// This is the signature creation time. As a default upper bound, use the current
+	// time.
+	validityTime := time.Now()
+
+	bundleVerified, err = VerifyBundle(ctx, sig, co, co.RekorClient)
 	if err != nil {
 		return false, fmt.Errorf("error verifying bundle: %w", err)
 	}
 
-	// If the --offline flag was specified, fail here
+	if bundleVerified {
+		// Update with the verified bundle's integrated time.
+		validityTime, err = getBundleIntegratedTime(sig)
+		if err != nil {
+			return false, fmt.Errorf("error getting bundle integrated time: %w", err)
+		}
+	}
+
+	// If the --offline flag was specified, fail here. bundleVerified returns false with
+	// no error when there was no bundle provided.
+	// TODO: You can be offline when you are verifying with a public key. This shouldn't
```

Comment:
Could you clarify what's meant by this?

Reply:
Will also follow-up in the re-review of the verification main function.

But this means that: if you are verifying with a normal PKI system, and using --offline, then you MIGHT not require timestamping/tlog verification. In that case, we might not want to return an error. I'm not sure.

Reply:
If you have a key, then timestamping isn't required, but tlog verification is still useful to know that the artifact is auditable. If you have a cert, then both timestamping and auditable is useful. Not sure if that helps haha.

Reply:
I think I agree -- I think the TODO can be removed then, will remove in the fix-up. I'll try to update the top-level verification logic comments as well.